### ISSUE TITLE
sbt 0.13.13 does not copy src/test/resources to test-classes during test:compile

### ISSUE THREAD

**Issue by NAJ10** [sbt/sbt#2841]
Unpack the attached sbtissue.zip, unzip it and cd to sbtissue dir. 
sbt clean test:compile
Under 0.13.12 this will copy the file src/test/resources/a.properties to target/scala-2.11/test-classes
Change project/build.properties to use 0.13.13 and repeat sbt clean test:compile
Under 0.13.13 the file a.properties is not copied to target/scala-2.11/test-classes

Under 0.13.12 it gives

```
ls -l target/scala-2.11/test-classes/
total 12
-rw-rw-r-- 1 n n   17 Nov 22 14:55 a.properties
-rw-rw-r-- 1 n n 1437 Nov 22 15:18 TestClass.class
-rw-rw-r-- 1 n n 1742 Nov 22 15:18 TestClass$.class
```

Under 0.13.13 it gives

```
ls -l target/scala-2.11/test-classes/
total 8
-rw-rw-r-- 1 n n 1437 Nov 22 15:19 TestClass.class
-rw-rw-r-- 1 n n 1742 Nov 22 15:19 TestClass$.class
```

The problem seems to me to be that going from 0.13.12 to 0.13.13 changes the behaviour of what is copied to the test-classes directory. 0.13.12 copies the contents of src/test/resources but 0.13.13 does not.

I would expect 0.13.13 to behave the same as 0.13.12 with respect to the behaviour of copying the contents of src/test/resources to target/scala-2.11/test-classes.

For what it is worth when you use Maven to compile the same project it copies the contents of src/test/resources to target/test-classes
[sbtissue.zip](https://github.com/sbt/sbt/files/606761/sbtissue.zip)

**Comment by eed3si9n** [comment]
The behavior you described actually works as designed. I understand that sbt 0.13.13 may looked as if it regressed and in fact it's cause by some other bug that will be fixed 0.13.14, but in general I don't think we can guarantee that resource copying happens for `compile` task in any configuration.
For that please use `test:product`. It will causing additional zipping, but if it's easier to remember you can also use `test:package` task.

**Comment by DavidMecham** [comment]
If thats as designed its a bad design.
